src/pages/en/guides/rss.mdx

```diff
@@ -34,115 +34,156 @@ First, install `@astrojs/rss` using your preferred package manager:
+:::
+
+`@astrojs/rss` also exposes an `rssSchema` for use with collection schemas. This ensures your frontmatter contains all properties expected by an RSS feed item. You can apply this schema to your content config like so:
+
+```js title="src/content/config.ts" "rssSchema"
+import { rssSchema } from '@astrojs/rss';
+
+const blog = defineCollection({
+  schema: rssSchema,
 });
+
+export const collections = { blog };
 ```
 
-See [Vite's glob import documentation](https://vitejs.dev/guide/features.html#glob-import) for more on this import syntax.
+### Using glob imports
 
-### 2. List of RSS feed objects
+You may generate an RSS feed from documents in `src/pages/`. Since `Astro.glob` is not available from endpoint routes, we've included a `pagesGlobToRssItems()` helper. This accepts an `import.meta.glob` result ([see Vite documentation](https://vitejs.dev/guide/features.html#glob-import)) and outputs an array of valid RSS feed items.
 
-We recommend this option for `.md` files outside of the `pages` directory. This is common when generating routes [via `getStaticPaths`](/en/reference/api-reference/#getstaticpaths).
+You can apply `pagesGlobToRssItems()` like so. This function assumes that you are globbing for items inside `src/pages/`, and all necessary feed properties are present in each document's frontmatter.
 
-For instance, say your `.md` posts are stored under a `src/posts/` directory. Each post has a `title`, `pubDate`, and `slug` in its frontmatter, where `slug` corresponds to the output URL on your site. We can generate an RSS feed using [Vite's `import.meta.glob` helper](https://vitejs.dev/guide/features.html#glob-import) like so:
+```js title="src/pages/rss.xml.js" "pagesGlobToRssItems" "await pagesGlobToRssItems("
+import rss, { pagesGlobToRssItems } from '@astrojs/rss';
 
-```js
-// src/pages/rss.xml.js
-import rss from '@astrojs/rss';
-
-const postImportResult = import.meta.glob('../posts/**/*.md', { eager: true });
-const posts = Object.values(postImportResult);
-
-export const get = () => rss({
-  title: 'Buzz’s Blog',
-  description: 'A humble Astronaut’s guide to the stars',
-  site: import.meta.env.SITE,
-  items: posts.map((post) => ({
-    link: post.url,
-    title: post.frontmatter.title,
-    pubDate: post.frontmatter.pubDate,
-  }))
-});
+export async function get(context) {
+  return rss({
+    title: 'Buzz’s Blog',
+    description: 'A humble Astronaut’s guide to the stars',
+    site: context.site,
+    items: await pagesGlobToRssItems(
+      import.meta.glob('./blog/*.{md,mdx}'),
+    ),
+  });
+}
 ```
 
 ### Including full post content
 
 <Since v="1.6.14" />
 
-By default, the Astro RSS integration does not support including the content of each of your posts in the feed itself. 
+Items also accept a `content` key containing the full content of the post as HTML.
```

Comment:
```suggestion
The `content` key contains the full content of the post as HTML. This allows you to make your entire post content available to RSS feed readers.
```
If we've mentioned the `content` property above, I think we can refer to it as a known/assumed thing in this intro.